daemon/src/engine/server/CryptoChallege.h

```diff
@@ -0,0 +1,633 @@
+/*
```

Comment:
Or you can do:

```cpp
using Data = std::vector<uint8_t>;
bool DecryptBase64(const Data& input, const Data& key, Data* output) {
    Data rawInput;
    if (!Base64Decode(input, &rawInput)) {
        return false;
    }
    Data decryptedHex;
    if (!Decrypt(rawInput, input, &decryptedHex)) {
        return false;
    }
    return HexDecode(decryptedHex, output)
}
```
The code will end up being shorter and easier to understand, it's a double win.